src/backend/access/index/amapi.c

```diff
@@ -20,7 +20,18 @@
 #include "utils/builtins.h"
 #include "utils/syscache.h"
 
+is_likeam_hook_type is_same_index_am_hook = NULL;
 
+bool
+isIndexAccessMethod(Oid relam, Oid indexAccessMethod)
+{
+	if ((is_same_index_am_hook && (*is_same_index_am_hook)(indexAccessMethod, relam)) ||
+		(!is_same_index_am_hook && relam == indexAccessMethod))
+	{
+		return true;
+	}
+	return false;
+}
```

Comment:
Could this simply to:
```c
if (relam == indexAccessMethod)
  return true;
return (is_same_index_am_hook && (*is_same_index_am_hook)(indexAccessMethod, relam);
```
The key difference is: 
relam is  indexAccessMethod and is_same_index_am_hook is not null, could it happen? And which one  has the priority?

Reply:
Theoretically，when is_same_index_am_hook is not null, it should include relam == indexAccessMethod, but it's hard to make decision for user, so I think is_same_index_am_hook has higher priority.